src/patternfly/components/Toolbar/toolbar.scss

```diff
@@ -221,30 +235,36 @@ $pf-v6--include-toolbar-breakpoints: true !default;
 
 // - Toolbar group
 .#{$toolbar}__group {
-  row-gap: var(--#{$toolbar}__group--RowGap, var(--#{$toolbar}--RowGap));
-  column-gap: var(--#{$toolbar}__group--ColumnGap, var(--#{$toolbar}--ColumnGap));
-  align-items: baseline;
-
-  // - Toolbar icon button group - Toolbar filter group
-  &.pf-m-icon-button-group {
-    column-gap: var(--#{$toolbar}__group--m-icon-button-group--ColumnGap, 0);
-  }
+  row-gap: var(--#{$toolbar}__group--RowGap);
+  column-gap: var(--#{$toolbar}__group--ColumnGap);
 
+  // - Toolbar filter group
   &.pf-m-filter-group {
     column-gap: var(--#{$toolbar}__group--m-filter-group--ColumnGap);
```

Comment:
This might be a little nit and I think this would actually be another core issue for the filter group component directly but, the semantic gap spacer will need updated to match this screenshot. visually it looks right, just needs the updated semantic naming i believe 
<img width="638" alt="Screenshot 2024-07-03 at 4 29 07 PM" src="https://github.com/patternfly/patternfly/assets/150823809/d1926aff-6b98-4ccc-a835-2aee3a23f821">